pyomo/core/expr/numvalue.py

```diff
@@ -264,15 +264,6 @@ def polynomial_degree(obj):
 # constants get repeated many times.  KnownConstants lets us re-use /
 # share constants we have seen before.
 #
-# Note:
-#   For now, all constants are coerced to floats.  This avoids integer
-#   division in Python 2.x.  (At least some of the time.)
-#
-#   When we eliminate support for Python 2.x, we will not need this
-#   coercion.  The main difference in the following code is that we will
-#   need to index KnownConstants by both the class type and value, since
-#   INT, FLOAT and LONG values sometimes hash the same.
-#
```

Comment:
I think we should keep this comment unless we can confirm with @jsiirola that it isn't needed. Line 305 below still seems to be casting things to floats and this comment claims that this isn't needed.

Reply:
I will say, I tried to remove the casting to float and it caused failures. So I can undo the comment, but I will also note that the comment is misleading and should be edited sooner rather than later.